### PR TITLE
Core/Battleground: Extra check on disabled battleground

### PR DESCRIPTION
**Pull request by Golrag**
**Changes proposed:**

-  Add extra check when queueing for a battleground to see if it is disabled with hotfixes.
-  Also define some BattlemasterList Flags.

**Target branch(es):**

- master

**Issues addressed:** none?

**Tests performed:**
- Builds
- Tested if the enabled battlegrounds are still queue-able

**Known issues and TODO list:** (add/remove lines as needed)

- Would be great to get some information about the unknown flags.

### CODE REVIEW

File: src/server/game/DataStores/DBCEnums.h [hunk 1]
```diff
@@ -168,6 +168,16 @@ enum ArtifactPowerFlag : uint8
 
 #define BATTLE_PET_SPECIES_MAX_ID 2796
 
+enum BattlemasterListFlags
+{
+    BATTLEMASTER_LIST_FLAG_DISABLED             = 0x01,
+    BATTLEMASTER_LIST_FLAG_UNK02                = 0x02,
```
Comment (Golrag) [inline src/server/game/DataStores/DBCEnums.h]: Seems like this flag allows you to queue without selected roles?

Comment (Shauren) [reply src/server/game/DataStores/DBCEnums.h]: This is correct

**Comment by Shauren** [comment]
Additionally BATTLEMASTER_LIST_FLAG_FACTIONAL = 0x40 (alliance vs horde, affects scoreboard tabs display)

File: src/server/game/DataStores/DBCEnums.h [hunk 2]
```diff
@@ -168,6 +168,16 @@ enum ArtifactPowerFlag : uint8
 
 #define BATTLE_PET_SPECIES_MAX_ID 2796
 
+enum BattlemasterListFlags
+{
+    BATTLEMASTER_LIST_FLAG_DISABLED             = 0x01,
+    BATTLEMASTER_LIST_FLAG_SKIP_ROLE_CHECK      = 0x02,
+    BATTLEMASTER_LIST_FLAG_FACTIONAL            = 0x04,
```
Comment (Shauren) [inline src/server/game/DataStores/DBCEnums.h]: 0x40, not 0x04

Comment (Golrag) [reply src/server/game/DataStores/DBCEnums.h]: oh woops

**Review by Shauren — APPROVED** [review]

**Pull request merged**